test/e2e/inventory_v1beta2_grpc_test.go

```diff
@@ -803,3 +803,298 @@ func enableShortMode(t *testing.T) {
+				LocalResourceId: resourceId,
+				ApiHref:         "https://api.example.com/hosts/checkbulk-host-0001",
+				ConsoleHref:     proto.String("https://console.example.com/hosts/checkbulk-host-0001"),
+			},
+			Common: &structpb.Struct{
+				Fields: map[string]*structpb.Value{
+					"workspace_id": structpb.NewStringValue(trueWorkspace),
+				},
+			},
+			Reporter: reporterStruct,
+		},
+	}
+	_, err = client.ReportResource(ctx, req)
+	assert.NoError(t, err, "Failed to Report Resource")
+
+	// Build CheckBulk with one expected TRUE and one expected FALSE
+	checkBulkReq := &pbv1beta2.CheckBulkRequest{
+		Items: []*pbv1beta2.CheckBulkRequestItem{
+			{
+				Object: &pbv1beta2.ResourceReference{
+					ResourceType: "host",
+					ResourceId:   resourceId,
+					Reporter: &pbv1beta2.ReporterReference{
+						Type:       reporterType,
+						InstanceId: proto.String(reporterInstanceId),
+					},
+				},
+				Relation: "workspace",
+				Subject: &pbv1beta2.SubjectReference{
+					Resource: &pbv1beta2.ResourceReference{
+						ResourceType: "workspace",
+						ResourceId:   trueWorkspace,
+						Reporter: &pbv1beta2.ReporterReference{
+							Type: "rbac",
+						},
+					},
+				},
+			},
+			{
+				Object: &pbv1beta2.ResourceReference{
+					ResourceType: "host",
+					ResourceId:   resourceId,
+					Reporter: &pbv1beta2.ReporterReference{
+						Type:       reporterType,
+						InstanceId: proto.String(reporterInstanceId),
+					},
+				},
+				Relation: "workspace",
+				Subject: &pbv1beta2.SubjectReference{
+					Resource: &pbv1beta2.ResourceReference{
+						ResourceType: "workspace",
+						ResourceId:   falseWorkspace,
+						Reporter: &pbv1beta2.ReporterReference{
+							Type: "rbac",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	// Poll up to 10 seconds to account for eventual consistency
+	observed := false
+	for i := 0; i < 10; i++ {
+		resp, err := client.CheckBulk(ctx, checkBulkReq)
+		if err == nil && resp != nil && len(resp.GetPairs()) == 2 {
+			pairs := resp.GetPairs()
+			gotTrue := pairs[0].GetItem() != nil && pairs[0].GetItem().GetAllowed() == pbv1beta2.Allowed_ALLOWED_TRUE
+			gotFalse := pairs[1].GetItem() != nil && pairs[1].GetItem().GetAllowed() == pbv1beta2.Allowed_ALLOWED_FALSE
+			if gotTrue && gotFalse {
+				observed = true
+				break
+			}
+			t.Logf("CheckBulk attempt %d: got allowed = [%v, %v], expected [TRUE, FALSE]", i+1, pairs[0].GetItem().GetAllowed(), pairs[1].GetItem().GetAllowed())
+		} else if err != nil {
+			t.Logf("CheckBulk request failed (attempt %d): %v", i+1, err)
+		}
+		if i < 9 {
+			time.Sleep(1 * time.Second)
+		}
+	}
+	assert.True(t, observed, "CheckBulk didn't return [TRUE, FALSE] within timeout")
+
+	// Cleanup
+	delReq := &pbv1beta2.DeleteResourceRequest{
+		Reference: &pbv1beta2.ResourceReference{
+			ResourceType: "host",
+			ResourceId:   resourceId,
+			Reporter: &pbv1beta2.ReporterReference{
+				Type:       reporterType,
+				InstanceId: proto.String(reporterInstanceId),
+			},
+		},
+	}
```

Comment:
We should consider putting any cleanup in the `defer` so that if tests fail they do not bleed into others